main.py

```diff
@@ -33,11 +34,14 @@ def __init__(self, input_lines: List[str]):
         self.result = ParsingResult()
         self.complete_parse()
 
+        self.versions_parser = VersionsParser(self.lines)
+
     def complete_parse(self):
         # todo: :)
         pass
     
     def get_results(self) -> ParsingResult:
+        self.result.versions = self.versions_parser.get_versions()
         return self.result

```

Comment:
The parsing call itself should happen in a method `complete_parse`, so that we can return using `get_results` quickly and multiple times, if neccesary.

```suggestion
    def complete_parse(self):
        self.result.versions = self.versions_parser.get_versions()
        pass
    
    def get_results(self) -> ParsingResult:
        return self.result

```